core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -74,7 +75,7 @@
     public static final String SAFE_PARAMETERS_SYSTEM_PROPERTY_NAME = ParametersAction.class.getName() +
             ".safeParameters";
 
-    private transient List<String> safeParameters;
+    private Set<String> safeParameters;
```

Comment:
0) The fix gets into 2.7
1) User installs 2.5.1 LTS with the fix (just an example version)
2) Then he installs 2.0 to get a missing feature
3) After installation he gets issues from the OldData monitor

It's not a blocker for merge, this fix is valuable in any case